packages/bootstrap/extra/tests/test_bootstrap.py

```diff
@@ -14,60 +14,57 @@
 log = logging.getLogger(__name__)
 log.setLevel(logging.DEBUG)
 
+zk_hosts = '127.0.0.1:2181'
 
-class TestBootstrap():
-    def setup_class(self):
-        self.tmpdir = os.path.abspath('tmp')
-        os.makedirs(self.tmpdir, exist_ok=True)
 
-        self.zk_container_name = ''.join(random.choice(string.ascii_uppercase + string.digits) for _ in range(6))
-        subprocess.check_call(['docker', 'run', '-d', '-p', '2181:2181', '-p',
-                               '2888:2888', '-p', '3888:3888', '--name', self.zk_container_name, 'jplock/zookeeper'])
-        self.zk_hosts = '127.0.0.1:2181'
+@pytest.fixture(scope='function')
+def zk_server(tmpdir):
+    zk_container_name = ''.join(random.choice(string.ascii_uppercase + string.digits) for _ in range(6))
+    # TODO(cmaloney): Add a python context manager for dockerized daemons
+    subprocess.check_call(['docker', 'run', '-d', '-p', '2181:2181', '-p',
+                           '2888:2888', '-p', '3888:3888', '--name', zk_container_name, 'jplock/zookeeper'])
 
-        conn_retry_policy = KazooRetry(max_tries=-1, delay=0.1, max_delay=0.1)
-        cmd_retry_policy = KazooRetry(max_tries=3, delay=0.3, backoff=1, max_delay=1, ignore_expire=False)
-        zk = KazooClient(hosts=self.zk_hosts, connection_retry=conn_retry_policy, command_retry=cmd_retry_policy)
-        zk.start()
+    conn_retry_policy = KazooRetry(max_tries=-1, delay=0.1, max_delay=0.1)
+    cmd_retry_policy = KazooRetry(max_tries=3, delay=0.3, backoff=1, max_delay=1, ignore_expire=False)
+    zk = KazooClient(hosts=zk_hosts, connection_retry=conn_retry_policy, command_retry=cmd_retry_policy)
+    zk.start()
 
-        children = zk.get_children('/')
-        for child in children:
-            if child == 'zookeeper':
-                continue
-            zk.delete('/' + child, recursive=True)
+    children = zk.get_children('/')
+    for child in children:
+        if child == 'zookeeper':
+            continue
+        zk.delete('/' + child, recursive=True)
 
-        self.zk = zk
+    yield zk
 
-    def teardown_class(self):
-        self.zk.stop()
-        self.zk.close()
-        subprocess.check_call(['docker', 'rm', '-f', self.zk_container_name])
+    zk.stop()
```

Comment:
nit: should probably try moving kazoo to have a context manager interface option